plugins/outputs/file/file.go

```diff
@@ -19,12 +20,15 @@ import (
 var sampleConfig string
 
 type File struct {
-	Files               []string        `toml:"files"`
-	RotationInterval    config.Duration `toml:"rotation_interval"`
-	RotationMaxSize     config.Size     `toml:"rotation_max_size"`
-	RotationMaxArchives int             `toml:"rotation_max_archives"`
-	UseBatchFormat      bool            `toml:"use_batch_format"`
-	Log                 telegraf.Logger `toml:"-"`
+	Files                []string        `toml:"files"`
+	RotationInterval     config.Duration `toml:"rotation_interval"`
+	RotationMaxSize      config.Size     `toml:"rotation_max_size"`
+	RotationMaxArchives  int             `toml:"rotation_max_archives"`
+	UseBatchFormat       bool            `toml:"use_batch_format"`
+	CompressionAlgorithm string          `toml:"compression_algorithm"`
+	CompressionLevel     int             `toml:"compression_level"`
+	encoder              internal.ContentEncoder
```

Comment:
Could you move the `encoder` to the section where the `writer`, `closers` and `serializer` also are?